src/cswinrt/code_writers.h

```diff
@@ -9969,17 +9969,26 @@ public static IntPtr Make()
 }
 
 static readonly % _factory = new %();
-public static ObjectReference<I> ActivateInstance<
-#if NET5_0_OR_GREATER
-    [DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicFields)]
+
+#if NET6_0_OR_GREATER
+[RequiresUnreferencedCode("This method is not trim-safe, and is only supported for use when not using trimming (or AOT).")]
+[Obsolete("This method is deprecated and will be removed in a future release.")]
+[EditorBrowsable(EditorBrowsableState.Never)]
 #endif
-    I>()
+public static ObjectReference<I> ActivateInstance<I>()
 {
     IntPtr instance = _factory.ActivateInstance();
 
     return ObjectReference<IInspectable.Vftbl>.Attach(ref instance, global::WinRT.Interop.IID.IID_IInspectable).As<I>();
 }
 
+public static ObjectReference<I> ActivateInstance<I>(Guid iid)
```

Comment:
Is anyone calling this?

Reply:
No, it'd be a new API. Just figured it might make sense to add so people using the current one would have something else to migrate to that's safe and not obsolete? Though if you don't think anyone would be using either one we can also just remove both of them directly. Honestly I was wondering if we could do that in the first place.